compiler/AST/primitive.cpp

```diff
@@ -997,6 +997,7 @@ initPrimitive() {
   prim_def(PRIM_GET_USER_LINE, "_get_user_line", returnInfoDefaultInt, true, true);
   prim_def(PRIM_GET_USER_FILE, "_get_user_file", returnInfoInt32, true, true);
 
+  prim_def(PRIM_PROXY_REF, "proxy reference", returnInfoVoid, false);
```

Comment:
Can you put a comment here describing the purpose and usage of this new primitive ?

Reply:
I see now it has to do with resolution order for constrained generics. Can you rename it to something that indicates that better? E.g. `PRIM_RESOLVE_IN_ORDER` ?